tests/integration/managedclusteraddon_test.go

```diff
@@ -0,0 +1,146 @@
+limitations under the License.
+*/
+
+package integration_test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	tokenexchange "github.com/red-hat-storage/odf-multicluster-orchestrator/addons/token-exchange"
+	multiclusterv1alpha1 "github.com/red-hat-storage/odf-multicluster-orchestrator/api/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	addonapiv1alpha1 "open-cluster-management.io/api/addon/v1alpha1"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+)
+
+var (
+	mirrorPeer1 = multiclusterv1alpha1.MirrorPeer{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "mirrorpeer1",
+		},
+	}
+	mirrorPeer1LookupKey = types.NamespacedName{Namespace: mirrorPeer1.Namespace, Name: mirrorPeer1.Name}
+	mcAddOn1LookupKey    = types.NamespacedName{Namespace: "cluster1", Name: tokenexchange.TokenExchangeName}
+	mcAddOn2LookupKey    = types.NamespacedName{Namespace: "cluster2", Name: tokenexchange.TokenExchangeName}
+)
+
+var _ = Describe("ManagedClusterAddOn creation, updation and deletion", func() {
+	When("creating or updating ManagedClusterAddOn", func() {
+		It("should not return validation error", func() {
+			newMirrorPeer := mirrorPeer1.DeepCopy()
+			newMirrorPeer.Spec = multiclusterv1alpha1.MirrorPeerSpec{
+				Items: []multiclusterv1alpha1.PeerRef{
+					{
+						ClusterName: "cluster1",
+						StorageClusterRef: multiclusterv1alpha1.StorageClusterRef{
+							Name:      "test-storagecluster",
+							Namespace: "test-namespace",
+						},
+					},
+					{
+						ClusterName: "cluster2",
+						StorageClusterRef: multiclusterv1alpha1.StorageClusterRef{
+							Name:      "test-storagecluster",
+							Namespace: "test-namespace",
+						},
+					},
+				},
+			}
+			var mcAddOn1 addonapiv1alpha1.ManagedClusterAddOn
+			var mcAddOn2 addonapiv1alpha1.ManagedClusterAddOn
+			managedcluster1 := clusterv1.ManagedCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster1",
+				},
+			}
+			managedcluster2 := clusterv1.ManagedCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster2",
+				},
+			}
+			ns1 := v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster1",
+				},
+			}
+			ns2 := v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster2",
+				},
+			}
+			By("creating MirrorPeer. Also, this should automatically create ManagedClusterAddOn", func() {
+				err := k8sClient.Create(context.TODO(), &managedcluster1, &client.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				err = k8sClient.Create(context.TODO(), &managedcluster2, &client.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				err = k8sClient.Create(context.TODO(), &ns1, &client.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				err = k8sClient.Create(context.TODO(), &ns2, &client.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred())
```

Comment:
These should be done in `Before()` block. It'll be called just before running your tests.